src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +230,214 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
+    String refreshTokenExpiration = tokens.getString(REFRESH_TOKEN_EXPIRATION);
+    // Use the ResponseBuilder rather than RMB-generated code. We need to do this because
+    // RMB generated-code does not allow multiple headers with the same key -- which is what we need
+    // here.
+    var body = new JsonObject()
+        .put(ACCESS_TOKEN_EXPIRATION, accessTokenExpiration)
+        .put(REFRESH_TOKEN_EXPIRATION, refreshTokenExpiration)
+        .toString();
+    return Response.status(201)
+        .header(SET_COOKIE, refreshTokenCookie(refreshToken, refreshTokenExpiration))
+        .header(SET_COOKIE, accessTokenCookie(accessToken, accessTokenExpiration))
+        .type(MediaType.APPLICATION_JSON)
+        .entity(body)
+        .build();
+  }
+
+  private Response logoutResponse() {
+    // Use the ResponseBuilder rather than RMB-generated code. We need to do this because
+    // RMB generated-code does not allow multiple headers with the same key -- which is what we need
+    // here.
+    return Response.status(204)
+        .header(SET_COOKIE, REFRESH_TOKEN + "=; Path=/authn; Expires=" + COOKIE_EXPIRES_FOR_DELETE)
+        .header(SET_COOKIE, ACCESS_TOKEN + "=; Expires=" + COOKIE_EXPIRES_FOR_DELETE)
+        .build();
+  }
+
+  private Response tokenResponseLegacy(JsonObject tokenJson) {
+    String authToken = tokenJson.getString("token");
+    var response = new LoginResponse().withOkapiToken(authToken);
+    return PostAuthnLoginResponse.respond201WithApplicationJson(response,
+        PostAuthnLoginResponse.headersFor201().withXOkapiToken(authToken));
+  }
+
+  private void handleLogout(String cookieHeader, Map<String, String> okapiHeaders,
+  Handler<AsyncResult<Response>> asyncResultHandler) {
+    try {
+      logger.debug("Cookie in handleLogout is: {}", cookieHeader);
+
+      String tenantId = getTenant(okapiHeaders);
+      String okapiURL = okapiHeaders.get(XOkapiHeaders.URL);
+      String accessToken = okapiHeaders.get(XOkapiHeaders.TOKEN);
+
+      Future<HttpResponse<Buffer>> logoutFuture;
+      if (cookieHeader == null) {
+        logoutFuture = logoutAll(tenantId, okapiURL, accessToken);
+      } else {
+        var p = new TokenCookieParser(cookieHeader);
+        logoutFuture = logout(tenantId, okapiURL, accessToken, p.refreshToken);
+      }
+
+      logoutFuture.onSuccess(r -> {
+      if (r.statusCode() >= 400) {
+          logger.error("{} (status code: {})", TOKEN_LOGOUT_UNPROCESSABLE, r.statusCode());
+          asyncResultHandler.handle(Future.succeededFuture(
+            PostAuthnLogoutResponse.respond422WithApplicationJson(getErrors(
+              TOKEN_LOGOUT_UNPROCESSABLE, TOKEN_LOGOUT_UNPROCESSABLE_CODE))));
+          return;
+        }
+
+        asyncResultHandler.handle(Future.succeededFuture(logoutResponse()));
+      });
+
+      logoutFuture.onFailure(e -> {
+        logger.error("{}: {} {}", INTERNAL_ERROR, e.getMessage(), e.getStackTrace());
```

Comment:
logger automatically prints the stacktrace if the last argument is a Throwable - leave out the last {} and replace e.getStackTrace() with e:

```
       logger.error("{}: {}", INTERNAL_ERROR, e.getMessage(), e);
```

The same multiple times below.

Reply:
Great to know, thanks.